functions/slackEvents.js

```diff
@@ -167,49 +191,86 @@ async function sendMessageToSlackResponseUrl(responseUrl, message) {
   });
 }
 
-async function emailShoppingList(ticketID, userID) {
-  const records = await getRecordsWithTicketID(INTAKE_TABLE, ticketID);
+function validateVolunteerAssigned(records, volunteerKey) {
+  const mismatchedVolunteerTickets = _.filter(records, ([, fields,]) => {
+    return !_.includes(fields.deliveryVolunteer, volunteerKey);
+  });
+  if (!_.isEmpty(mismatchedVolunteerTickets)) {
+    const mismatchedTicketIDs = _.join(
+      _.map(mismatchedVolunteerTickets, ([, fields,]) => {
+        return fields.ticketID;
+      }),
+      ', '
+    );
+    throw new Error(`Volunteer ${volunteerKey} requested a shopping list for tickets they are not assigned to: ${mismatchedTicketIDs}`);
+  }
+}
+
+function renderTicketMetadata(fields) {
+  return `
+**Ticket ID:** ${fields.ticketID}<br/>
```

Comment:
should we consider turning this stuff into a template?

Reply:
I was trying not to spend too long on it last night, but yes we should.